src/tx_builder/mod.rs

```diff
@@ -349,6 +416,187 @@ impl CapacityBalancer {
             force_small_change_as_fee: None,
         }
     }
+
+    pub fn new_with_provider(fee_rate: u64, capacity_provider: CapacityProvider) -> Self {
+        CapacityBalancer {
+            fee_rate: FeeRate::from_u64(fee_rate),
+            capacity_provider,
+            change_lock_script: None,
+            force_small_change_as_fee: None,
+        }
+    }
+    pub fn new_with_provider_max_fee(
+        fee_rate: u64,
+        capacity_provider: CapacityProvider,
+        max_fee: u64,
+    ) -> Self {
```

Comment:
`CapacityBalancer` has only 4 fields, why not just use `CapacityBalancer` directory to construct it?